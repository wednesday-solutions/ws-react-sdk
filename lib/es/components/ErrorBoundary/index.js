import"../_rollup-plugin-inject-process-env-1259e43d.js";import r from"react";import e from"prop-types";import{translate as o}from"../IntlGlobalProvider/index.js";import"react-intl";class t extends r.Component{constructor(r){super(r),this.state={hasError:!1,error:null}}static getDerivedStateFromError(r){return{hasError:!0,error:r}}componentDidCatch(r,e){console.error(r,e)}render(){return this.state.hasError?r.createElement("h1",null,o("something_went_wrong")):this.props.children}}t.propTypes={children:e.oneOfType([e.arrayOf(e.node),e.node])};export default t;
