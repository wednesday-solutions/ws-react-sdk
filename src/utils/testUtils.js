import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { DEFAULT_LOCALE, translationMessages } from '@src/i18n';
import IntlGlobalProvider from '@components/IntlGlobalProvider';

export const renderWithIntl = children => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );
};

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export default {
  timeout,
  renderWithIntl,
  getComponentStyles,
  apiResponseGenerator
};
