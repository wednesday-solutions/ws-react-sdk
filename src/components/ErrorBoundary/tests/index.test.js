import React from 'react';
import { renderWithIntl } from '@utils/testUtils';
import ErrorBoundary from '../index';

const Something = () => null;

describe('ErrorBoundary', () => {
  it('should match the snapshot', () => {
    const { wrapper } = renderWithIntl(
      <ErrorBoundary>
        <Something />
      </ErrorBoundary>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should throw Error', () => {
    const Throws = () => {
      throw new Error('Oh no!');
    };
    const { getAllByTestId } = renderWithIntl(
      <ErrorBoundary>
        <Throws />
      </ErrorBoundary>
    );
    expect(getAllByTestId('error')).toMatchSnapshot();
    expect(Throws).toThrow('Oh no!');
  });
});
