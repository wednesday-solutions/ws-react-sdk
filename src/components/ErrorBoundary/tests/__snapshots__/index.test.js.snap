// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should match the snapshot 1`] = `undefined`;

exports[`ErrorBoundary should throw Error 1`] = `
Array [
  <h1
    data-testid="error"
  >
    something_went_wrong
  </h1>,
]
`;
